src/spring_downloader.js

```diff
@@ -105,9 +105,9 @@ class SpringDownloader extends EventEmitter {
 		currentDownloader = prdDownloader;
 	}
 
-	downloadResource(resource) {
+	downloadResource(name, resource) {
```

Comment:
This function is also called from `src/launcher_wizard.js` and you aren't fixing it there. In addition, you won't be able to call it from there correctly because resources in the config at the moment don't have names: https://github.com/beyond-all-reason/BYAR-Chobby/blob/master/dist_cfg/config.json#L21-L33.

Also below it could invoke `PrdDownloader.downloadResource` so you would have to change type signature there too as it doesn't have the name as argument at the mmoment.

Maybe instead of taking the name as argument, name should be just property of the resource that defaults to destination in emitter?

Reply:
Ok, i just spent some time to understand the origin of configs and their resulting resource downloads...

Instead of adding a name property to every config, which doesn't add new information, I'll try to use destination as name.